src/dns/nm-dns-manager.c

```diff
@@ -1637,28 +1637,25 @@ _resolvconf_resolved_managed (void)
 		"/run/systemd/resolve/resolv.conf",
 		"/lib/systemd/resolv.conf",
 		"/usr/lib/systemd/resolv.conf",
+		NULL
 	};
-	GFile *f;
-	GFileInfo *info;
-	gboolean ret = FALSE;
+	struct stat etc_resconf_st;
+	const char* p;
 
-	f = g_file_new_for_path (_PATH_RESCONF);
-	info = g_file_query_info (f,
-	                          G_FILE_ATTRIBUTE_STANDARD_IS_SYMLINK","\
-	                          G_FILE_ATTRIBUTE_STANDARD_SYMLINK_TARGET,
-	                          G_FILE_QUERY_INFO_NOFOLLOW_SYMLINKS,
-	                          NULL, NULL);
+	if (stat (_PATH_RESCONF, &etc_resconf_st) != 0)
+		return FALSE;
 
-	if (info && g_file_info_get_is_symlink (info)) {
-		ret = nm_utils_strv_find_first ((gchar **) resolved_paths,
-		                                G_N_ELEMENTS (resolved_paths),
-		                                g_file_info_get_symlink_target (info)) >= 0;
-	}
+	for (p = resolved_paths[0]; *p; p++) {
+		struct stat systemd_resconf_st;
 
-	g_clear_object(&info);
-	g_clear_object(&f);
+		if (stat (p, &systemd_resconf_st) != 0)
+			continue;
```

Comment:
I think it's sufficient to add `After=systemd-resolved.service` to `NetworkManager.service`. This will guarantee that NM will only be started after resolved has signalled to systemd that it is ready to process requests. (It does *not* have the effect of causing a request to start NM to also start resolved--that's done with the `Wants=` or `Requires=` directive, which we don't use).

I can add back the check for `/etc/resolv.conf` being a symlink itself, although it is not strictly necessary, and without the check the code even works for the edge case where the administrator unintentionally hardlinked `/etc/resolv.conf` to `/lib/systemd/resolv.conf`.